src/python/WMCore/WMSpec/Steps/Executors/LogArchive.py

```diff
@@ -47,6 +46,56 @@ def pre(self, emulator=None):
         logging.info("Steps.Executors.LogArchive.pre called")
         return None
 
+    def sendLogToEOS(self, overrides, tarBallLocation, useNewStageOutCode):
+        """
+        :param overrides: dictionary for setting the eos lfn.
+        :return: None (copy log archive to eos location.)
+        Don'f fail the job if copy fails
+        """
+        eosStageOutParams = {}
+        eosStageOutParams['command'] = "xrdcp"
+        eosStageOutParams['option'] = "--wma-disablewriterecovery"
+        eosStageOutParams['phedex-node'] = overrides.get('eos-phedex-node', "T2_CH_CERN")
+        eosStageOutParams['lfn-prefix'] = overrides.get('eos-lfn-prefix',
+                                                        "root://eoscms.cern.ch//eos/cms/store/logs/prod/recent")
+
+        if not eosStageOutParams['lfn-prefix']:
+            # if overrides for eos-lfn-prefix is set to None or "", don't copy the log to eos
+            return
+
+        if not useNewStageOutCode:
+            # old style
+            eosmanager = StageOutMgr.StageOutMgr(**eosStageOutParams)
+            eosmanager.numberOfRetries = self.step.retryCount
+            eosmanager.retryPauseTime = self.step.retryDelay
+        else:
+            # new style
+            logging.info("LOGARCHIVE IS USING NEW STAGEOUT CODE For EOS Copy")
+            eosmanager = WMCore.Storage.FileManager.StageOutMgr(
+                retryPauseTime=self.step.retryDelay,
+                numberOfRetries=self.step.retryCount,
+                **eosStageOutParams)
+
+        eosFileInfo = {'LFN': self.getEOSLogLFN(),
+                       'PFN': tarBallLocation,
+                       'PNN': None,
+                       'GUID': None
+                       }
+
+        try:
+            eosmanager(eosFileInfo)
+            eosServerPrefix = eosStageOutParams['lfn-prefix'].replace("root://eoscms.cern.ch//eos/cms",
+                                                                      "https://eoscmshttp.cern.ch")
+            self.report.setLogURL(eosServerPrefix + eosFileInfo['LFN'])
+            self.report.persist("Report.pkl")
+        except Alarm:
+            msg = "Indefinite hang during stageOut of logArchive to EOS, ignoring it"
+            logging.error(msg)
+        except Exception as ex:
+            logging.exception("EOS copy failes lfn: %s. Error: %s" % (eosFileInfo['LFN'], str(ex)))
```

Comment:
Seangchan, don't hate me :-) But can you fix this pylint warning too:
 W1201, line 95 in LogArchive.sendLogToEOS: Specify string format arguments as logging function parameters 
?
basically updating this line to
`logging.exception("EOS copy failes lfn: %s. Error: %s", eosFileInfo['LFN'], str(ex))`

Reply:
updated failes --> failed

Reply:
I hate you Alan. 😄 Actually thank you very much for the review. I always forget that.
Anyway, I still a problem of setting the url in fwjr. I will fix other pylint errors when that is fixed.